testing/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/BuildOperationNotificationsFixture.groovy

```diff
@@ -49,12 +48,18 @@ class BuildOperationNotificationsFixture {
         }
     }
 
+    /**
+     * @see org.gradle.internal.operations.notify.BuildOperationNotificationBridge
+     */
     static String injectNotificationListenerBuildLogic() {
+        // The registration of the listener must happen before the BuildOperationNotificationValve is closed preemptively
+        // The actual Develocity plugin register a listener before `projectsLoaded`.
+        // Here, we rely on the order of `projectLoaded` callback registrations.
         """
-            rootProject {
-                if(gradle.isRootBuild()) {
+            gradle.projectsLoaded {
```

Comment:
❓ When is `projectsLoaded` fired? Is that happening after all `settingsEvaluated` callbacks are done?

Reply:
@wolfs Not sure about `all settingsEvaluated callbacks are done` which I assume means "every build settings evaluated", but for each particular build, `projectsLoaded` happens after `settingsEvaluated`

Reply:
There should only be one Settings involved that allows the Develocity plugin to check in, right? And as long as we wait long enough there that init scripts can apply the Develocity plugin in a settings evaluated, then that is good enough.